### PR TITLE
[dhcp6] add multicast solicit and accept more general replies

### PR DESCRIPTION
**Pull request by jrodriguez42**
Adding a compiler flag to enable OpenThread devices to transmit multicast DHCPv6 Solicit messages as well as adding code to enable compatibility with DNSmasq DHCPv6 Server replies.

### CODE REVIEW

**Comment by jrodriguez42** [comment]
Sorry about that, I used the GitHub API to update my fork and it didn't use my correct email

File: src/core/net/dhcp6_client.cpp [hunk 1]
```diff
@@ -275,12 +275,23 @@ otError Dhcp6Client::Solicit(uint16_t aRloc16)
     SuccessOrExit(error = AppendIaAddress(*message, aRloc16));
     SuccessOrExit(error = AppendRapidCommit(*message));
 
+#if OPENTHREAD_ENABLE_DHCP6_MULTICAST_SOLICIT
+    messageInfo.GetPeerAddr().mFields.m16[0] = HostSwap16(0xff03);
+    messageInfo.GetPeerAddr().mFields.m16[1] = HostSwap16(0x0000);
```
Comment (jwhui) [inline src/core/net/dhcp6_client.cpp]: nit: probably unnecessary to explicitly zero out these fields - the [MessageInfo constructor](https://github.com/openthread/openthread/blob/80f982301ca199edfacdd8176f037822a3d144fc/src/core/net/socket.hpp#L62) zeros these fields out.

File: src/core/net/dhcp6_client.cpp [hunk 2]
```diff
@@ -431,6 +442,10 @@ void Dhcp6Client::ProcessReply(Message &aMessage)
     uint16_t offset = aMessage.GetOffset();
     uint16_t length = aMessage.GetLength() - aMessage.GetOffset();
     uint16_t optionOffset;
+    if ((optionOffset = FindOption(aMessage, offset, length, kOptionStatusCode)) > 0)
```
Comment (jwhui) [inline src/core/net/dhcp6_client.cpp]: style: add empty line above

**Comment by jwhui** [comment]
@jrodriguez42, this PR looks good but is [failing the code style check](https://travis-ci.org/openthread/openthread/jobs/571572912). The patch is provided in the log. Otherwise, you can also use clang-format to automatically reformat the code (see [STYLE_GUIDE.md](https://github.com/openthread/openthread/blob/master/CONTRIBUTING.md#coding-conventions-and-style) for more details).

**Comment by jrodriguez42** [comment]
Sorry about that. I realized I pushed some changes that caused another check to fail and it took me a few tries to roll it back and properly fix it.

**Comment by jwhui** [comment]
@jrodriguez42, thanks for the updates. I think you accidentally added `FETCH_HEAD` and `dnsmasq_dhcp6_server`. Can you remove those from this PR?

**Pull request merged**